### ISSUE TITLE
Sync BatchNorm parameters between GPU's

### ISSUE THREAD

**Issue by landersson** [facebook/fbcunn#104]
I have a model trained using DataParallel on 3 GPUS. When running in test mode, this model will occasionally produce different end results for identical input data, depending on which GPU the input sample happens to get evaluated on. I think I have narrowed this down to BatchNormalization layers having ended up with slightly different mean and variance parameters during training, and these have not been synchronised/averaged out between the GPU's during training. I suppose this is not a major issue in general, but it makes it hard to verify that other modifications have not introduced errors and to accurately compare models in torch and other frameworks to each other. 

I'm thinking about applying a quick fix for this that will help my use case by simply adding a method to the DataParallel class to copy the parameters of any BN layers from the "homeDevice" GPU to the other GPU's and call that after loading up the model. I guess the proper solution would be to "mix" and sync parameters during training, but I'm mainly aiming to use already trained models in this case. Either way, any guidance on how to best implement this would be appreciated.
